crml/transaction-payment/src/lib.rs

```diff
@@ -917,30 +916,26 @@ mod tests {
 			});
 	}
 
+	// For the () implmentation of NegativeImbalance, we reduce the total issuance by the fee
 	#[test]
-	fn fee_exchange_temporary_storage_unused_for_normal_calls() {
+	fn imbalanced_is_executed_for_fee_payment() {
+		let base_fee = 5;
 		ExtBuilder::default()
-			.base_fee(5)
+			.base_fee(base_fee)
 			.balance_factor(1000)
 			.build()
 			.execute_with(|| {
-				let len: u64 = 10;
-
-				// `GAS_FEE_EXCHANGE_KEY` temporary storage should be unset
-				let fee_exchange = FeeExchange::new_v1(VALID_ASSET_TO_BUY_FEE, 111);
-				let transaction_payment_with_fee_exchange =
-					ChargeTransactionPayment::<Runtime>::from(0, Some(fee_exchange.clone()));
-				assert!(transaction_payment_with_fee_exchange
-					.pre_dispatch(&1, CALL, info_from_weight(3), len as usize)
+				let len: u64 = 0;
+				let total_issuance = Balances::total_issuance();
+				let dispatch_info = info_from_weight(1);
+				let fee = ChargeTransactionPayment::<Runtime>::compute_fee(0, dispatch_info.clone(), 0);
+
+				let transaction_payment = ChargeTransactionPayment::<Runtime>::from(0, None);
+				assert!(transaction_payment
+					.pre_dispatch(&1, CALL, dispatch_info, len as usize)
 					.is_ok());
-				assert!(storage::unhashed::get::<Option<()>>(&GAS_FEE_EXCHANGE_KEY).is_none());
```

Comment:
hmm, can't see why this test was removed?

Reply:
Yeah, that doesn't seem right, nice catch

Reply:
Added it back, thanks @holygits